homeassistant/components/onvif/camera.py

```diff
@@ -142,16 +142,20 @@ async def async_camera_image(
 
         if self.device.capabilities.snapshot:
             try:
-                image = await self.device.device.get_snapshot(
+                if image := await self.device.device.get_snapshot(
                     self.profile.token, self._basic_auth
-                )
-                return image
+                ):
+                    return image
             except ONVIFError as err:
                 LOGGER.error(
                     "Fetch snapshot image failed from %s, falling back to FFmpeg; %s",
                     self.device.name,
                     err,
                 )
+            LOGGER.error(
+                "Fetch snapshot image failed from %s, falling back to FFmpeg",
+                self.device.name,
+            )
```

Comment:
Otherwise, we end up with possibly two errors being logged.

```suggestion
            else:
                LOGGER.error(
                    "Fetch snapshot image failed from %s, falling back to FFmpeg",
                    self.device.name,
                )
```